hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/RewardCalculator.java

```diff
@@ -20,46 +20,53 @@
  * ‍
  */
 
+import com.hedera.services.ledger.properties.AccountProperty;
 import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.state.merkle.MerkleNetworkContext;
 import com.hedera.services.state.merkle.MerkleStakingInfo;
 import com.hedera.services.utils.EntityNum;
 import com.swirlds.merkle.map.MerkleMap;
+import org.apache.commons.lang3.tuple.Pair;
 
 import javax.inject.Inject;
 import java.time.LocalDate;
 import java.time.ZoneId;
+import java.util.Map;
 import java.util.function.Supplier;
+;
+import static com.hedera.services.ledger.interceptors.StakeChangeManager.finalBalanceGiven;
+import static com.hedera.services.ledger.properties.AccountProperty.BALANCE;
+import static com.hedera.services.ledger.properties.AccountProperty.STAKE_PERIOD_START;
 
 public class RewardCalculator {
-	public static final EntityNum stakingFundAccount = EntityNum.fromLong(800L);
-	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts;
 	private final Supplier<MerkleMap<EntityNum, MerkleStakingInfo>> stakingInfo;
+	private final Supplier<MerkleNetworkContext> networkCtx;
+	private long rewardsPaid;
 
 	public static final ZoneId zoneUTC = ZoneId.of("UTC");
 
 	@Inject
-	public RewardCalculator(final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts,
-			final Supplier<MerkleMap<EntityNum, MerkleStakingInfo>> stakingInfo) {
-		this.accounts = accounts;
+	public RewardCalculator(
+			final Supplier<MerkleMap<EntityNum, MerkleStakingInfo>> stakingInfo,
+			final Supplier<MerkleNetworkContext> networkCtx) {
 		this.stakingInfo = stakingInfo;
+		this.networkCtx = networkCtx;
 	}
 
-	public final long computeAndApplyRewards(final EntityNum accountNum) {
+	public final Pair<Long, Long> computeRewards(final MerkleAccount account) {
 		long todayNumber = LocalDate.now(zoneUTC).toEpochDay();
```

Comment:
I would go ahead and extract the logic to determine the current `stakePeriod` into a separate `StakePeriodManager`.